scripts/zones/VeLugannon_Palace/npcs/qm3.lua

```diff
@@ -17,6 +17,7 @@ entity.onTrade = function(player, npc, trade)
             player:messageSpecial(ID.text.KNIFE_CHANGES_SHAPE, xi.items.BUCCANEERS_KNIFE)
             npcUtil.giveItem(player, xi.items.BARTHOLOMEWS_KNIFE) -- Btm. Knife
         else
+            player:tradeComplete(false)
```

Comment:
If you fail you will need to steal a new knife to try again; a failed knife will never transform (verified 25/05/2020).

is this an era change or are we allowing them to keep the knife even if they fail?

Reply:
Ok, updated PR to use the worn item mechanism to limit to only one trade per knife.